ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/ChainSel.hs

```diff
@@ -538,6 +555,14 @@ chainSelectionForBlock cdb@CDB{..} blockCache hdr punish = electric $ do
         traceWith addBlockTracer (TryAddToCurrentChain p)
         addToCurrentChain succsOf' curChainAndLedger maxExtra
 
+      -- The block @b@ fits onto the end of our current chain, but we can't add
+      -- it because the LoE limit was exceeded, so we store it in @cdbLoEBlocks@
+      -- for later reprocessing.
+      | pointHash tipPoint == headerPrevHash hdr -> do
+        atomically (modifyTVar cdbLoEBlocks (hdr :))
+        traceWith addBlockTracer (PostponedByLoE p)
+        return tipPoint
```

Comment:
This currently does not handle cases where blocks arrive out of order, ie we select `hdr`, but not some of its successors already in the VolDB. Idea: use `VolatileDB.filterByPredecessor`/`succsOf` to find all successors of the new tip, and add them to `cdbLoEBlocks`.

Reply:
✅

Reply:
a026aaafb2db062bd763fb48e1f41e851ca9c75c

Reply:
✅